### PR TITLE
.github: update CODEOWNERs

### PR DESCRIPTION
**Pull request by tbg**
I went through "recent commits" via

```

files=()
for f in $(git log -n 100 --pretty="format:" --name-only); do
    if [ -e "${f}" ]; then files+=("${f}"); fi
done
go run ./pkg/cmd/whoownsit ${files[@]} > out.txt
```

and looked at the files owned by @cockroachdb/kv and
@cockroachdb/test-eng in order to motivate some tweaks to the rules.

This should hopefully make the notifications more relevant.

Release note: None

### CODE REVIEW

File: .github/CODEOWNERS
Comment (stevendanna) [inline .github/CODEOWNERS:235]: This seems like it would be pretty noisy for dev-inf.  Every dependency change in a package can result in a BUILD.bazel update from `make bazel-generate`

Comment (otan) [reply .github/CODEOWNERS:235]: BUILD.bazel to me should be owned by the team that directory belongs to.
they should be modifying, e.g., the test dependencies in future, not dev-inf.

File: .github/CODEOWNERS
Comment (otan) [inline .github/CODEOWNERS:222]: do we want to review changes to `testutils` and `testutilsccl`? i was thinking about removing it today because theyre pretty noisy and usually not relevant

File: .github/CODEOWNERS
Comment (tbg) [reply .github/CODEOWNERS:235]: It's a bit awkward - I don't think we want to request reviews based on changes to BUILD.bazel. Sure, teams kind of own them, but don't really know how to make manual changes to them. If they make manual changes, surely dev-inf should take a look. I agree that it is bad that all codegen-updates ping dev-inf, so best I can come up with is `dev-inf-noreview`. We could also make them unowned which has sort of the same effect in practice.

The reason I don't want to own them to the teams is that any time Ricky goes and changes something, the team doesn't care - dev-inf cares - and yet the team gets a ping. Maybe this is a small enough issue and we can drop this line, but there will be situations where all bazel files change at once.

File: .github/CODEOWNERS
Comment (tbg) [reply .github/CODEOWNERS:222]: That's fine by me, I was trying to be surgical because I don't really know what else is in that package. Your suggestion is to make that `-noreview`?

Comment (otan) [reply .github/CODEOWNERS:222]: yep!

File: .github/CODEOWNERS
Comment (otan) [reply .github/CODEOWNERS:235]: i think in that case we should do `-no-review`, at least for now.
i can buy the education argument :) one day i do imagine it belongs to the team

File: .github/CODEOWNERS
Comment (tbg) [inline .github/CODEOWNERS:238]: @otan do you have any insight on that? Must be a bug in our code, no?

Comment (tbg) [reply .github/CODEOWNERS:238]: For more context, `BUILD.bazel` as a rule doesn't work either - I *think* it was supposed to - that's why I started on `**/BUILD.bazel` in the first place.

Comment (otan) [reply .github/CODEOWNERS:240]: Probably the library we use in codeowners: "github.com/zabawaba99/go-gitignore"

**Review by otan — APPROVED** [review]

**Comment by tbg** [comment]
bors r=otan

**Comment by tbg** [comment]
logictestccl

bors r=otan

**Pull request merged**
